lightblue/src/main/java/org/esbtools/eventhandler/lightblue/client/BulkLightblueRequester.java

```diff
@@ -373,14 +375,36 @@ public U get() throws InterruptedException, ExecutionException {
             return result;
         }
 
-        // TODO(ahenning): This ignores the timeout because we aren't doing work in another thread
-        // I'm not sure that anyone will ever care, but if they did, we would do the same lazy
-        // stuff, but just do it in another thread which we could interrupt if it took too long.
-        // In that case we may have to build in interrupt checking during request processing.
         @Override
-        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException,
-                TimeoutException {
-            return get();
+        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+            if (cancelled) {
```

Comment:
There's some logic that's unfortunately duplicated in this method and `get()`. I would consider a scheme to try and consolidate them. Perhaps having a private `get(Optional<Duration> timeout)` that both public get methods delegate to would be a way to go, where you do all of the same logic except when it comes to calling the completer function, we could check if there is a timeout provided and either use another thread as below or just call the completer in the same thread as above.